subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -458,30 +67,11 @@ int lwm2m_firmware_cancel_transfer(void)
 
 int lwm2m_firmware_start_transfer(char *package_uri)
 {
-	/* close old socket */
-	if (firmware_ctx.sock_fd > -1) {
-		lwm2m_engine_context_close(&firmware_ctx);
-	}
+	req.write_cb = lwm2m_firmware_get_write_cb();
 
-	(void)memset(&firmware_ctx, 0, sizeof(struct lwm2m_ctx));
-	firmware_ctx.sock_fd = -1;
-	firmware_ctx.fault_cb = socket_fault_cb;
-	firmware_retry = 0;
+	/* start file transfer work */
+	lwm2m_pull_context_start_transfer(package_uri, req, K_NO_WAIT);
```

Comment:
I think this object should implement the same logic for handling start_transfer failures as SW mgmt.

Reply:
done